consensus/istanbul/backend/announce.go

```diff
@@ -1289,15 +1311,23 @@ func (sb *Backend) sendEnodeCertificateMsg(peer consensus.Peer, msg *istanbul.Me
 	return peer.Send(istanbul.EnodeCertificateMsg, payload)
 }
 
-func (sb *Backend) setEnodeCertificateMsg(msg *istanbul.Message) error {
-	sb.enodeCertificateMsgMu.Lock()
-	var enodeCertificate enodeCertificate
-	if err := rlp.DecodeBytes(msg.Msg, &enodeCertificate); err != nil {
+func (sb *Backend) SetEnodeCertificateMsg(msg *istanbul.Message) error {
+	var newEnodeCertificate istanbul.EnodeCertificate
+	if err := rlp.DecodeBytes(msg.Msg, &newEnodeCertificate); err != nil {
 		return err
 	}
-	sb.enodeCertificateMsg = msg
-	sb.enodeCertificateMsgVersion = enodeCertificate.Version
-	sb.enodeCertificateMsgMu.Unlock()
+
+	sb.enodeCertificateMsgMu.Lock()
+	defer sb.enodeCertificateMsgMu.Unlock()
+
+	// Already have a more recent or the same enodeCertificate
+	if newEnodeCertificate.Version <= sb.enodeCertificateMsgVersion {
+		return nil
```

Comment:
It seems odd to me that is this not an error.

Reply:
Agreed.  Fixed.